### ISSUE TITLE
Switch to use cloud-based solution for create MacOS releases

### ISSUE THREAD

**Issue by tylerslaton** [acorn-io/runtime#1632]
Right now our current solution for releasing Mac binaries is not scalable. We should switch to a more comprehensive solution like [MacStadium](https://www.macstadium.com/bare-metal-mac).

**Comment by cjellick** [comment]
Don't do anything destructive before @ibuildthecloud gets back from PTO

**Comment by tylerslaton** [comment]
Fixed by #1880

**Comment by tylerslaton** [comment]
If we ever come back to this and need som context - we have switch to using an open-source project called `apple-codesign` (and its binary, `rcodesign`). It is written in Rust and docs for it can be found [here](https://gregoryszorc.com/docs/apple-codesign/0.17.0/apple_codesign_rcodesign.html). Originally we were creating a `DMG` and notarizating/stapling it using Mac's propriety software and packaging format. However, since we mainly support the `wget` and `brew` installation methods (neither of which required notarization or stapling), we decided to turn off the creation of the `DMG. 

Should we ever need the `DMG`-like flow for whatever reason, we can enable it [here](https://github.com/acorn-io/runtime/blob/bc1f0c316409f733c1c3a34fa2af43361f561d2c/.goreleaser.yml#L40). It was updated (in the PR linked above) to instead be a `ZIP` format which is much easier to properly create in a Linux environment.

After all of this was said and done we no longer need to use a pet-mac and do everything in a buildjet runner now.
